File: tests/fixtures/project_java/src/main/java/auth/Verify.java
package auth;

import java.util.Map;

public class Verify {

    public static Map<String, Object> verifyToken(String token) {
        if (token == null || token.isEmpty()) {
            throw new IllegalArgumentException("token is required");
        }

        Map<String, Object> payload = Jwt.decodeToken(token);
        if (Jwt.isTokenExpired(payload)) {
            throw new SecurityException("token has expired");
        }

        return payload;
    }
}

File: tests/fixtures/project_rust/src/auth/verify.rs
use std::collections::HashMap;
use crate::auth::jwt;
use crate::utils::errors::AppError;

pub fn verify_token(token: &str) -> Result<HashMap<String, String>, AppError> {
    if token.is_empty() {
        return Err(AppError::new("token is required"));
    }

    let payload = jwt::decode_token(token)?;
    // Check expiry if present
    if jwt::is_token_expired(None) {
        return Err(AppError::new("token has expired"));
    }

    Ok(payload)
}
